nil pointer dereference in 3.9.0 applicationserver

<!--
Thanks for submitting a bug report. Please fill the template below,
otherwise we will not be able to process this bug report.
-->

#### Summary
<!-- Summarize the problem in a few sentences: -->

TTN stack 3.9.0 is currently crash-looping with a nil pointer dereference.

#### Steps to Reproduce
<!-- How can we reproduce the problem? -->

1. Start the Stack.

<!--
Please upload relevant configuration (as .txt).
If you use the command "ttn-lw-stack config", you can redact sensitive config.
-->

#### What do you see now?
<!-- Please paste terminal output, upload logs (as .txt) or upload screenshots. -->
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x18d2aba]
goroutine 124 [running]:
go.thethings.network/lorawan-stack/v3/pkg/applicationserver.(*ApplicationServer).decryptDownlinkMessage(0xc000843a20, 0x2774260, 0xc000d68c90, 0xc0007baec0, 0x1b, 0xc000581e38, 0x8, 0x0, 0xc000581e40, 0xc000581e48, ...)
	/Users/veg/devel/lorawan-stack-backend/pkg/applicationserver/applicationserver.go:1029 +0x18a
go.thethings.network/lorawan-stack/v3/pkg/applicationserver.(*ApplicationServer).handleUp(0xc000843a20, 0x2774260, 0xc000d68c30, 0xc000227300, 0xc000646200, 0x5f2eda2b, 0xa92c0f0, 0x3b4e166d88bc6)
	/Users/veg/devel/lorawan-stack-backend/pkg/applicationserver/applicationserver.go:540 +0x637
go.thethings.network/lorawan-stack/v3/pkg/applicationserver.(*link).sendUp(0xc000646200, 0x2774260, 0xc000470b10, 0xc000227300, 0xc000825d28, 0xc000646200, 0x2792d40)
	/Users/veg/devel/lorawan-stack-backend/pkg/applicationserver/linking.go:348 +0x2c6
go.thethings.network/lorawan-stack/v3/pkg/applicationserver.(*ApplicationServer).link(0xc000843a20, 0x2774260, 0xc000470870, 0xc000945300, 0x8, 0x0, 0xc00094c4c0, 0x0, 0x0)
	/Users/veg/devel/lorawan-stack-backend/pkg/applicationserver/linking.go:258 +0x9bc
go.thethings.network/lorawan-stack/v3/pkg/applicationserver.(*ApplicationServer).startLinkTask.func1(0x2774260, 0xc000470870, 0x39b0920, 0x0)
	/Users/veg/devel/lorawan-stack-backend/pkg/applicationserver/linking.go:63 +0x229
go.thethings.network/lorawan-stack/v3/pkg/component.(*Component).StartTask.func1(0xc0004708a0, 0x2774260, 0xc000470870, 0x279e120, 0xc00094c0c0, 0x2, 0x38e0cd0, 0x3, 0x3, 0x3fb999999999999a)
	/Users/veg/devel/lorawan-stack-backend/pkg/component/tasks.go:82 +0x84
created by go.thethings.network/lorawan-stack/v3/pkg/component.(*Component).StartTask
	/Users/veg/devel/lorawan-stack-backend/pkg/component/tasks.go:77 +0x150
```
#### What do you want to see instead?
<!-- Please add some examples or mock-ups if applicable. -->
No crash

#### Environment
<!--
Your environment: OS/Browser/Gateway/Device/...? Versions? IDs/EUIs?
Paste the output of "ttn-lw-cli version" or "ttn-lw-stack version" if applicable.
-->

Version 3.9.0 running in Kubernetes
a handful of Laird gateways connecting via basicstation and 3 end devices (xdot based). It suddenly started happening after I upgraded to 3.9.0 from 3.8.7.

#### How do you propose to implement this?
<!-- Please think about how this could be fixed. -->
There appears to be no check for a non-existent session in applicationserver.go:

line 1022:
```
var session *ttnpb.Session
	switch {
	case dev.Session != nil && bytes.Equal(dev.Session.SessionKeyID, msg.SessionKeyID):
		session = dev.Session
	case dev.PendingSession != nil && bytes.Equal(dev.PendingSession.SessionKeyID, msg.SessionKeyID):
		session = dev.PendingSession
	}
	if session.GetAppSKey() == nil {
		return errNoAppSKey.New()
	}
```

My naive fix, based on not understanding what could cause this is:
```
        if session == nil || session.GetAppSKey() == nil {
```

#### How do you propose to test this?
<!-- Please think about how this is verified as fixed. -->
Works On My Machine 

#### Can you do this yourself and submit a Pull Request?
<!-- You can also @mention experts if you need help with this. -->
Yes. But only if you think it's a good idea and I haven't missed the point entirely.

--- a/pkg/applicationserver/applicationserver.go
+++ b/pkg/applicationserver/applicationserver.go
@@ -1025,6 +1025,8 @@ func (as *ApplicationServer) decryptDownlinkMessage(ctx context.Context, ids ttn
 		session = dev.Session
 	case dev.PendingSession != nil && bytes.Equal(dev.PendingSession.SessionKeyID, msg.SessionKeyID):
 		session = dev.PendingSession
+	default:
+		return errUnknownSession.New()
 	}
 	if session.GetAppSKey() == nil {
 		return errNoAppSKey.New()

